pyiceberg/io/pyarrow.py

```diff
@@ -636,7 +636,13 @@ def visit_fixed(self, fixed_type: FixedType) -> pa.DataType:
         return pa.binary(len(fixed_type))
 
     def visit_decimal(self, decimal_type: DecimalType) -> pa.DataType:
-        return pa.decimal128(decimal_type.precision, decimal_type.scale)
+        return (
+            pa.decimal32(decimal_type.precision, decimal_type.scale)
+            if decimal_type.precision <= 9
+            else pa.decimal64(decimal_type.precision, decimal_type.scale)
+            if decimal_type.precision <= 18
+            else pa.decimal128(decimal_type.precision, decimal_type.scale)
```

Comment:
"""
Scale is fixed, precision must be 38 or less
"""
from https://iceberg.apache.org/spec/#primitive-types 

![Screenshot 2025-05-09 at 8 38 15 AM](https://github.com/user-attachments/assets/a27a4100-b12e-4ab8-bca9-6084b89a6b06)

Reply:
pyarrow.decimal128 supports up to precision 38
https://arrow.apache.org/docs/python/generated/pyarrow.decimal128.html#pyarrow-decimal128

Reply:
Thanks for checking. This inspired me to change the code to throw on precision >38.